### PR TITLE
fix(ssr): inline middleware during the build, not rely on file system

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

Closes https://github.com/withastro/astro/issues/8299

Reverts part of changes of https://github.com/withastro/astro/pull/8200

We can't load the middleware using a file system path because, in adapters like Vercel, the bundled files are eventually moved. For other adapters like cloudflare, that is not permitted.

## Testing

Current tests should pass.

I also tested locally and checked that the middleware is correctly bundled in SSR.

## Docs

N/A

### CODE REVIEW

File: packages/astro/src/core/build/plugins/plugin-manifest.ts [hunk 1]
```diff
@@ -237,9 +237,6 @@ function buildManifest(
 		// Set this to an empty string so that the runtime knows not to try and load this.
 		entryModules[BEFORE_HYDRATION_SCRIPT_ID] = '';
 	}
-	const isEdgeMiddleware =
-		// TODO: remove in Astro 4.0
-		settings.config.build.excludeMiddleware || settings.adapter?.adapterFeatures?.edgeMiddleware;
```
Comment (natemoo-re) [inline packages/astro/src/core/build/plugins/plugin-manifest.ts:242]: Did this logic get dropped? Looks like the `TODO` did, at least.

**Review by matthewp — APPROVED** [review]

File: packages/astro/src/core/build/static-build.ts [hunk 2]
```diff
@@ -123,14 +123,14 @@ export async function staticBuild(opts: StaticBuildOptions, internals: BuildInte
 		case settings.config.output === 'static': {
 			settings.timer.start('Static generate');
 			await generatePages(opts, internals);
-			await cleanServerOutput(opts);
+			// await cleanServerOutput(opts);
```
Comment (natemoo-re) [inline packages/astro/src/core/build/static-build.ts:126]: Debugging?

File: packages/astro/src/core/build/static-build.ts [hunk 3]
```diff
@@ -123,14 +123,14 @@ export async function staticBuild(opts: StaticBuildOptions, internals: BuildInte
...
 			settings.timer.end('Static generate');
 			return;
 		}
 		case isServerLikeOutput(settings.config): {
 			settings.timer.start('Server generate');
 			await generatePages(opts, internals);
-			await cleanStaticOutput(opts, internals);
+			// await cleanStaticOutput(opts, internals);
```
Comment (natemoo-re) [inline packages/astro/src/core/build/static-build.ts:133]: Debugging?

File: packages/integrations/vercel/test/edge-middleware.test.js [hunk 4]
```diff
@@ -19,17 +19,11 @@ describe('Vercel edge middleware', () => {
 	});
 
 	// TODO: The path here seems to be inconsistent?
-	it.skip('with edge handle file, should successfully build the middleware', async () => {
+	it.skip('without edge handle file, should successfully build the middleware', async () => {
```
Comment (natemoo-re) [inline packages/integrations/vercel/test/edge-middleware.test.js:22]: Still using `it.skip`, should it?

**Review by natemoo-re — COMMENTED** [review]
This seems reasonable! Still see a few changes to be cleaned up.

File: packages/integrations/vercel/test/edge-middleware.test.js (hunk 4 above)
Comment (ematipico) [reply packages/integrations/vercel/test/edge-middleware.test.js:22]: Yes. For some reasons that I don't understand, the path of the emitted files inside the fixtures is not predictable.

**Review by natemoo-re — APPROVED** [review]

**Pull request merged**